Nianjie/solution.py

```diff
@@ -0,0 +1,19 @@
+class Solution:
+    def trap(self, height: List[int]) -> int:
+        l = len(height)
+        if l == 0:
+            return 0
+        
+        left_max = [height[0]]
+        for i in range(l-1):
+            left_max.append(max(left_max[i], height[i+1]))
```

Comment:
The code works, but I think for easier readability we could have run indexes from 1 to l so that we deal with height[i]. Currently, the first impression it gives is that we are looking at next height